tslearn/tests/test_metrics.py

```diff
@@ -116,6 +116,32 @@ def test_masks():
                                [np.inf, np.inf, np.inf, np.inf, np.inf,  0.]])
     np.testing.assert_allclose(i_mask, reference_mask)
 
+    sz = 10
```

Comment:
Would there be a way to test that the output of a call with `metric_params` in there actually used those parameters (KMeans or KNN)? E.g. by checking which values are actually calculated in the matrix and comparing that to the calculated mask? The most simple way would be to pass metric_params with both `itakura_max_slope` and `sakoe_chiba_radius` and assert for Warning/Error

Reply:
Not sure how we could test that. Maybe ensure that we get a similar result as with Euclidean distance (when radius is minimal / slope constraint is the toughest possible)

Reply:
Yeah that is a nice idea, you would want to have a mask where only the diagonal is filled with 0's for that, right? Not sure how that can be created through `itakura_max_slope` or `sakoe_chiba_radius` (I'm not entirely familiar with those)

Reply:
Done.